lib/macho/macho_file.rb

```diff
@@ -170,26 +168,68 @@ def command(name)
 
     alias :[] :command
 
-    # Add a new load command to the Mach-O.
-    # @param lc [MachO::LoadCommand] the load command being added
+    # Replace a load command with another command in the Mach-O, preserving location.
+    # @param old_lc [MachO::LoadCommand] the load command being replaced
+    # @param new_lc [MachO::LoadCommand] the load command being added
     # @return [void]
     # @raise [MachO::HeaderPadError] if the new command exceeds the header pad buffer
-    # @note This is public, but methods like {#add_rpath} should be preferred.
-    def add_command(lc)
+    # @note This is public, but methods like {#dylib_id=} should be preferred.
+    def replace_command(old_lc, new_lc)
       context = LoadCommand::SerializationContext.context_for(self)
-      cmd_raw = lc.serialize(context)
+      cmd_raw = new_lc.serialize(context)
       new_sizeofcmds = sizeofcmds + cmd_raw.bytesize
-
       if header.class.bytesize + new_sizeofcmds > low_fileoff
         raise HeaderPadError.new(@filename)
       end
 
-      # update the raw data to reflect the new command
-      set_ncmds(ncmds + 1)
-      set_sizeofcmds(new_sizeofcmds)
-      @raw_data[header.class.bytesize + sizeofcmds, cmd_raw.bytesize] = cmd_raw
+      delete_command(old_lc)
+      insert_command(old_lc.view.offset, new_lc)
+    end
 
-      # synchronize fields with the raw data
+    # Add a new load command to the Mach-O.
+    # @param lc [MachO::LoadCommand] the load command being added
+    # @param options [Hash]
+    # @option options [Boolean] :repopulate (true) whether or not to repopulate
+    #  the instance fields
+    # @return [void]
+    # @note This is public, but methods like {#add_rpath} should be preferred.
+    #  Setting `repopulate` to false **will leave the instance in an
+    #  inconsistent state** unless {#populate_fields} is called **immediately**
+    #  afterwards.
+    def add_command(lc, options = {})
```

Comment:
We could, but my understanding of YARD is that you don't normally document an exception at a site unless that function throws it itself. Making an exception (no pun intended) here is reasonable though, since we don't expose `insert_command` as part of the public API.